coalib/results/TextPosition.py

```diff
@@ -7,20 +7,35 @@
 class TextPosition:
 
     @enforce_signature
-    def __init__(self, line: (int, None)=None, column: (int, None)=None):
+    def __init__(self,
+                 line: (int, None)=None,
+                 column: (int, None)=None,
+                 text: (tuple, list, None)=None,
```

Comment:
if the user gives position only, there's no way the other information can be derived. The whole point of having an object is so we can rely on it doing certain things for us, you remove that value and the object *may* be of no use for us and we'd need to do checks for every information which sucks. We have an object which is unusable in some cases.